x-pack/plugins/security/common/constants.ts

```diff
@@ -20,6 +20,7 @@ export const APPLICATION_PREFIX = 'kibana-';
 export const RESERVED_PRIVILEGES_APPLICATION_WILDCARD = 'kibana-*';
 
 export const AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER = 'auth_provider_hint';
+export const AUTH_URL_HASH_QUERY_STRING_PARAMETER = 'auth_url_hash';
```

Comment:
The significance of these was not clear to me until I started reading through the code. Perhaps some TSdoc comments would be helpful for folks who are looking at this in the future?
```suggestion
/**
 * This is the key of a query parameter that contains metadata about the selected authentication provider while the user is being redirected
 * during single-sign-on authentication flows. That query parameter is discarded after the authentication flow succeeds. See the
 * Authenticator, OIDCAuthenticationProvider, and SAMLAuthenticationProvider classes for more information.
 */
export const AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER = 'auth_provider_hint';
/**
 * This is the key of a query parameter that contains metadata about the (client-side) URL hash while the user is being redirected during
 * single-sign-on authentication flows. That query parameter is discarded after the authentication flow succeeds. See the Authenticator,
 * OIDCAuthenticationProvider, and SAMLAuthenticationProvider classes for more information.
 */
export const AUTH_URL_HASH_QUERY_STRING_PARAMETER = 'auth_url_hash';
```

Reply:
Agree, thanks!